CustomElementRenderer.js

```diff
@@ -30,6 +31,26 @@ export class CustomElementRender extends ElementRenderer {
     yield this.element.innerHTML;
   }
   *renderShadow() {
-    yield this.element.shadowRoot.innerHTML;
+    yield this.element.shadowRoot?.innerHTML;
   }
+
+  *renderSlots(slots) {
+		for (let [slot, value = ''] of Object.entries(slots)) {
+			if (slot !== 'default' && value) {
+				// Parse the value as a concatenated string
+				const fragment = parse5.parseFragment(`${value}`);
```

Comment:
could you explain why we need parse5 for this? Maybe a code example?

Reply:
It's more or less just a copy of the named slots handling in the official astro Lit plugin. ;)

I also do not like this mix of different redundant tools so much, but I used it as a first draft and placeholder to test the necessary mechanisms as easy as possible.

In the long run I try to use replaceable DOM adapters (most likely the [Environment modules](https://vitest.dev/guide/environment.html) from `vitest`) and realize all the necessary transformations just by DOM-API-call, which should work identical in all standard compliant stubs.